### PR TITLE
Added dock menu for macOS

### LINKED ISSUE #3024: Support dock menus
We should show at a minimum new window, new private window, and the list of windows there.
The list of windows would show the window title (i.e. active tab title)

Edit: Use `app.dock.setMenu(menu)`

### PR DESCRIPTION
**Pull request by NejcZdovc**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

## Description
Resolves #3024. As far as I can see the second part of this issue _the list of windows_ is already done.

Added doc menu for macOS with items:
- new window

## Auditors
@bbondy, @bsclifton

## Test Plan
Open Brave on macOS. Right click on the icon in the dock. There should be a menu item named "New window". Clicking on it should open a new Brave window.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
I would also like to add New private window, but as far as I can see Brave don't have this option. Any suggestions how to achieve this?

**Comment by Liunkae** [comment]
@NejcZdovc Works great, thanks! Since New Private Window is not implemented yet, I think it can be left out for now. (Tracked under #1288.)

File: app/browser/menu.js [hunk 1]
```diff
@@ -488,6 +489,16 @@ const createDebugSubmenu = () => {
   ]
 }
 
+const createDockSubmenu = () => {
+  return [
+    {
+      label: locale.translation('newWindow'),
+      accelerator: 'CmdOrCtrl+N',
```
Comment (darkdh) [inline app/browser/menu.js]: do we need accelerator for dock menu?

Comment (NejcZdovc) [inline app/browser/menu.js]: Sure, it's removed.

**Comment by darkdh** [comment]
good job! Thanks

**Comment by bsclifton** [comment]
Very cool- thanks so much :smile:

**Pull request merged**